python_scripts/repeatable/update_refids.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/python3
+# This script takes a CSV of archival object URIs as inputs, grabs the JSON data for each archival object, and updates
+# the archival objects' update_refid field to True, posting them back to ArchivesSpace which regenerates the refids.
+import argparse
+import os
+import sys
+
+from dotenv import load_dotenv, find_dotenv
+from loguru import logger
+from pathlib import Path
+
+sys.path.append(os.path.dirname('python_scripts'))  # Needed to import functions from utilities.py
+from python_scripts.utilities import ASpaceAPI, read_csv, write_to_file
+
+logger.remove()
+log_path = Path('../../logs', 'update_refids_{time:YYYY-MM-DD}.log')
+logger.add(str(log_path), format="{time}-{level}: {message}")
+
+# Find  and load environment-specific .env file
+env_file = find_dotenv(f'.env.{os.getenv("ENV", "dev")}')
+load_dotenv(env_file)
+
+def parseArguments():
+    """Parses the arguments fed to the script from the terminal or within a run configuration"""
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("csvPath", help="path to CSV input file", type=str)
+    parser.add_argument("-dR", "--dry-run", help="dry run?", action='store_true')
+    parser.add_argument("--version", action="version", version='%(prog)s - Version 1.0')
+
+    return parser.parse_args()
+
+
+def main(csv_path, dry_run=False):
+    """
+    This script takes a CSV of archival object URIs as inputs, grabs all the archival objects' JSON data using the API,
+    and updates the archival objects' update_refid field to True, posting them back to ArchivesSpace which regenerates
+    the refids.
+
+    The CSV should have the following data structure:
+    - Column 1 header = uri
+    - Column 1 rows = repositories/##/archival_objects/#######
+
+    Args:
+        csv_path (str): filepath of the Excel file with the agent IDs
+        dry_run (bool): if True, it prints the changed object_json but does not post the changes to ASpace
+    """
+    local_aspace = ASpaceAPI(os.getenv('as_api'), os.getenv('as_un'), os.getenv('as_pw'))
+    original_ao_json_data = str(Path('../../test_data', 'update_refids_original_data.jsonl'))
```

Comment:
Could this filepath also be set with an arg?  Or be created in the current directory?  Some other new directory?  I wouldn't think to look in `test_data` for anything to do with running a script.  In my mind `test_data` is just to hold data for test coverage.

Reply:
Changing it to be an argument is a good idea. I'll adjust that and resubmit.